### PR TITLE
Add username and email signup form

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Adds the username/email input screen and handles the trivial error (where field is blank).

The plan is to make a request to the service to have that validate our username and email, and then we can reflect those errors on the ui.

### CODE REVIEW

File: shared/reducers/signup.js [hunk 1]
```diff
@@ -5,25 +5,66 @@ import * as Constants from '../constants/signup'
...
+  phase: 'inviteCode'
 }
 
 export default function (state: SignupState = initialState, action: SignupActions): SignupState {
   switch (action.type) {
     case Constants.checkInviteCode:
-      // Assert this is the type we're looking at
-      var action: CheckInviteCode = action
-      if (action.error && action.error === true) {
+      if (action.error) {
```
Comment (MarcoPolo) [inline shared/reducers/signup.js]: Turns out we don't need to do `=== true`

File: shared/reducers/signup.js [hunk 2]
```diff
@@ -5,25 +5,66 @@ import * as Constants from '../constants/signup'
...
+  emailError: null,
+  passwordError: null,
+  phase: 'inviteCode'
 }
 
 export default function (state: SignupState = initialState, action: SignupActions): SignupState {
   switch (action.type) {
     case Constants.checkInviteCode:
-      // Assert this is the type we're looking at
-      var action: CheckInviteCode = action
```
Comment (MarcoPolo) [inline shared/reducers/signup.js]: This isn't needed (and actually get flow mad at you)

Comment (chrisnojima) [inline shared/reducers/signup.js]: well the names can't be the same

Comment (MarcoPolo) [inline shared/reducers/signup.js]: yeah, that was the problem

**Comment by chrisnojima** [comment]
:+1:

**Pull request merged**